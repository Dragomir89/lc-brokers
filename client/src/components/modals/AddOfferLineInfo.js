import React from 'react';

const AddOfferLineInfo = ({ label, value, dimension='' }) => {
    const style = !!value ? 'modal__info-position' : 'modal__info-position-danger'
    value = !!value ? `${value} ${dimension}` : '-- няма стойност --';
    
    return (
        <div className={style}>
            <p>{label}: </p>
            <p>{value}</p>
        </div>
    )
}

export default AddOfferLineInfo;