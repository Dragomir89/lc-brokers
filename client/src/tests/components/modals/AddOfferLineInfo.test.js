import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import '../../setupTests';
import AddOfferLineInfo from '../../../components/modals/AddOfferLineInfo';

test('should render correct info width data', () => {
    const wrapper = shallow(<AddOfferLineInfo label='Цена' value='200' dimension='лв'/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
});

test('should render correct info widthout value', () => {
    const wrapper = shallow(<AddOfferLineInfo label='Цена' />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});