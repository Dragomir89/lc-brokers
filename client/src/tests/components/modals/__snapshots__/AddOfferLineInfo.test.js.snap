// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct info width data 1`] = `
<div
  className="modal__info-position"
>
  <p>
    Цена
    : 
  </p>
  <p>
    200 лв
  </p>
</div>
`;

exports[`should render correct info widthout value 1`] = `
<div
  className="modal__info-position-danger"
>
  <p>
    Цена
    : 
  </p>
  <p>
    -- няма стойност --
  </p>
</div>
`;
